collector/src/main/java/com/bitmovin/analytics/stateMachines/PlayerState.java

```diff
@@ -124,13 +125,27 @@ void onExitState(PlayerStateMachine machine, long elapsedTime, PlayerState desin
     QUALITYCHANGE {
         @Override
         void onEnterState(PlayerStateMachine machine) {
+            machine.increaseQualityChangeCount();
+            if(!machine.isQualityChangeTimerRunning){
+                machine.qualityChangeResetTimeout.start();
+            }
         }
 
         @Override
         void onExitState(PlayerStateMachine machine, long elapsedTime, PlayerState destinationPlayerState) {
-            for (StateMachineListener listener : machine.getListeners()) {
-                listener.onQualityChange();
-            }
+           if(machine.isQualityChangeEventEnabled()) {
+               for (StateMachineListener listener : machine.getListeners()) {
+                   listener.onQualityChange();
+               }
+           }
+           else {
+               ErrorCode errorCode = new ErrorCode(10000, "ANALYTICS_QUALITY_CHANGE_THRESHOLD_EXCEEDED", null );
```

Comment:
Please create a constant for the message and error-code. I think we should already have something in place for the buffering -- if not, we will need it anyways.

Reply:
Created `AnalyticsErrorCodes` enum for quality change and buffering timeout errors, please check it out. This same class will also be in my PR for limiting duration of buffering event.